From 8089ff17bef875799eb66b985fe447aa8b651f62 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 05 Jan 2021 19:36:03 +0200
Subject: [PATCH] [SW-20719] preboot: add generic ram init hook

Add a generic RAM init hook which can be used to initialized any memory
types.
It is inteded to be used in Gaudi for initializing the HBM.
For now only move the SRAM scrambling code call to the new hook.

Change-Id: I8b392e013becb1622154d557e328750936aee1f9
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/goya2_arc_gnu/asic.c b/goya2_arc_gnu/asic.c
index 07fe02c..cb41d80 100644
--- a/goya2_arc_gnu/asic.c
+++ b/goya2_arc_gnu/asic.c
@@ -58,7 +58,7 @@
 }
 
 /* initialize sram scrambling before u-boot is downloaded */
-void sram_scrambling_init(void)
+static void sram_scrambling_init(void)
 {
 	regmap_writel(1, mmDCON0_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
 	regmap_writel(1, mmDCON0_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
@@ -69,3 +69,8 @@
 	regmap_writel(1, mmDCON3_HBW_RTR_IF0_RTR_H3_SCRAM_SRAM_EN);
 	regmap_writel(1, mmDCON3_HBW_RTR_IF1_RTR_H3_SCRAM_SRAM_EN);
 }
+
+void hl_ram_init(void)
+{
+	sram_scrambling_init();
+}
diff --git a/include/preboot.h b/include/preboot.h
index 231b44e..2b80c9e 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -142,6 +142,7 @@
 void preboot_finalize_enabled_caps(void);
 void preboot_conf_iatu_access(bool enable);
 void preboot_disable_iatu_access(void);
+void hl_ram_init(void);
 
 #endif /* __ASSEMBLY__ */
 
diff --git a/preboot/preboot.c b/preboot/preboot.c
index 063ae2c..ca625f7 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -206,8 +206,8 @@
 __weak void preboot_security_dynamic_img_close(void) {}
 __weak void preboot_security_cleanup_before_run(void) {}
 __weak void preboot_led_on(void) {}
-__weak void sram_scrambling_init(void) {}
 __weak void preboot_sp_sram_init(void) {}
+__weak void hl_ram_init(void) {}
 
 static bool preboot_is_overlap_address_range(u64 addr, int len)
 {
@@ -291,7 +291,7 @@
 	if (!is_sec_en)
 		hl_clock_gating_init();
 
-	sram_scrambling_init();
+	hl_ram_init();
 	preboot_dev_enable_caps();
 	module_handle_cmds(cmd_ops, ARRAY_SIZE(cmd_ops));
 }
